### PR TITLE
Deactivate the signal handler & activate Sentry on_crash callback

### PR DESCRIPTION
**Pull request by ChristopheLarchier**

### CODE REVIEW

File: src/libcommon/log/sentry/sentryhandler.cpp [hunk 1]
```diff
@@ -142,13 +149,17 @@ static sentry_value_t crashCallback(const sentry_ucontext_t *uctx, sentry_value_
     (void)uctx;
     (void)closure;
 
-    readObject(event);
+    std::cerr << "Sentry detected a crash in the app " << SentryHandler::appType() << std::endl;
 
-    // signum is unknown, all crashes will be considered as kills
-    KDC::SignalType signalType = KDC::fromInt<KDC::SignalType>(0);
-    std::cerr << "Server stopped with signal " << signalType << std::endl;
+    // signum is unknown, crashes will be considered as kills
```
Comment (luc-guyot-infomaniak) [inline src/libcommon/log/sentry/sentryhandler.cpp:154]:
```suggestion
    // As `signum` is unknown, a crash is considered as a kill.
```

File: src/libcommon/log/sentry/sentryhandler.cpp [hunk 2]
```diff
@@ -205,15 +233,12 @@ void SentryHandler::init(SentryProject project, int breadCrumbsSize) {
     sentry_options_set_debug(options, false);
     sentry_options_set_max_breadcrumbs(options, breadCrumbsSize);
 
-    bool isSet = false;
-#if defined(Q_OS_LINUX)
-    if (CommonUtility::envVarValue("KDRIVE_DEBUG_SENTRY_CRASH_CB", isSet); isSet) {
-        sentry_options_set_on_crash(options, crashCallback, NULL);
-    }
-    if (CommonUtility::envVarValue("KDRIVE_DEBUG_SENTRY_BEFORE_SEND_CB", isSet); isSet) {
+    // !!! Not Supported in Crashpad on macOS & Limitations in Crashpad on Windows for Fast-fail Crashes !!!
```
Comment (luc-guyot-infomaniak) [inline src/libcommon/log/sentry/sentryhandler.cpp:236]: **Nice to have**: a link to the doc.

File: src/libcommon/log/sentry/sentryhandler.cpp [hunk 3]
```diff
@@ -172,28 +189,39 @@ std::shared_ptr<SentryHandler> SentryHandler::instance() {
...
     }
 
+    _appType = appType;
+
+    bool isSet = false;
+    if (CommonUtility::envVarValue("KDRIVE_DEBUG_SENTRY_CRASH_CB", isSet); isSet) {
+        _debugCrashCallback = true;
+    }
+    isSet = false;
+    if (CommonUtility::envVarValue("KDRIVE_DEBUG_SENTRY_BEFORE_SEND_CB", isSet); isSet) {
```
Comment (luc-guyot-infomaniak) [inline src/libcommon/log/sentry/sentryhandler.cpp:212]: Maybe a comment about the fact that, if set, the second value will override the first one.

File: src/libcommon/log/sentry/sentryhandler.cpp [hunk 4]
```diff
@@ -374,6 +399,19 @@ void SentryHandler::updateEffectiveSentryUser(const SentryUser &user) {
     sentry_set_user(userValue);
 }
 
+void SentryHandler::writeEvent(const std::string &eventStr, bool crash) noexcept {
+    auto eventFilePath =
```
Comment (luc-guyot-infomaniak) [inline src/libcommon/log/sentry/sentryhandler.cpp:403]: Could be put into a function that would be then testable.

File: src/libcommon/utility/types.h [hunk 5]
```diff
@@ -100,7 +100,16 @@ static constexpr std::string_view clientCrashFileName(
 static constexpr std::string_view clientKillFileName(
         "kdrive_client.kill"); // Name of the file generated by the crash handler when a kill of the client occurs
 
-enum class AppType { Server, Client };
+static constexpr std::string_view serverCrashEventFileName(
```
Comment (luc-guyot-infomaniak) [inline src/libcommon/utility/types.h:103]: Why not defining these new strings and types where there are used?

File: src/libcommon/log/sentry/sentryhandler.cpp (hunk 1 above)
Comment (ChristopheLarchier) [reply src/libcommon/log/sentry/sentryhandler.cpp:154]: Done

File: src/libcommon/log/sentry/sentryhandler.cpp (hunk 2 above)
Comment (ChristopheLarchier) [reply src/libcommon/log/sentry/sentryhandler.cpp:236]: Done

File: src/libcommon/log/sentry/sentryhandler.cpp (hunk 3 above)
Comment (ChristopheLarchier) [reply src/libcommon/log/sentry/sentryhandler.cpp:212]: Done

File: src/libcommon/log/sentry/sentryhandler.cpp (hunk 4 above)
Comment (ChristopheLarchier) [reply src/libcommon/log/sentry/sentryhandler.cpp:403]: This function is called only for debugging... (when the concerned environment variables are set)

File: src/libcommon/utility/types.h (hunk 5 above)
Comment (ChristopheLarchier) [reply src/libcommon/utility/types.h:103]: This class is used in quite a few places (appclient, appserver, mainclient, mainserver, sentryhandler, utility). It was a duplicate, I just deleted the duplicate.

Comment (luc-guyot-infomaniak) [reply src/libcommon/utility/types.h:103]: Good to know, thanks! What about a namespace to keep the strings together?

File: src/libcommon/log/sentry/sentryhandler.cpp (hunk 4 above)
Comment (luc-guyot-infomaniak) [reply src/libcommon/log/sentry/sentryhandler.cpp:408]: Maybe it's overkill to write unit tests for debugging functions. So it's your call. Ideally, debugging code should work as well as production code (we don't want to debug debugging code when we need it work).

Comment (ChristopheLarchier) [reply src/libcommon/log/sentry/sentryhandler.cpp:409]: Done

File: src/libcommon/utility/types.h (hunk 5 above)
Comment (ChristopheLarchier) [reply src/libcommon/utility/types.h:104]: Done

**Review by luc-guyot-infomaniak — APPROVED** [review]

**Pull request merged**